File: android/app/src/main/java/com/otppoc/android/views/ContentView.kt
package com.otppoc.android.views

import androidx.compose.animation.AnimatedContent
import androidx.compose.animation.slideInHorizontally
import androidx.compose.animation.slideOutHorizontally
import androidx.compose.animation.togetherWith
import androidx.compose.animation.fadeIn
import androidx.compose.animation.fadeOut
import androidx.compose.foundation.background
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.rememberScrollState
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.foundation.verticalScroll
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.Shield
import androidx.compose.material3.Icon
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.runtime.collectAsState
import androidx.compose.runtime.getValue
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import androidx.lifecycle.viewmodel.compose.viewModel
import com.otppoc.android.ui.theme.AppColors
import com.otppoc.android.ui.theme.AppSpacing
import com.otppoc.android.viewmodels.AuthState
import com.otppoc.android.viewmodels.AuthViewModel
import com.otppoc.android.views.components.DefenseCard
import com.otppoc.android.views.components.StepIndicator

@Composable
fun ContentView(viewModel: AuthViewModel = viewModel()) {
    val state by viewModel.state.collectAsState()

    val currentStep = when (state) {
        is AuthState.PhoneInput, is AuthState.Error -> 0
        is AuthState.RunningPreflight, is AuthState.SolvingPoW,
        is AuthState.SendingOtp, is AuthState.OtpSent,
        is AuthState.VerifyingOtp -> 1
        is AuthState.Authenticated -> 2
    }

    Column(
        modifier = Modifier
            .fillMaxSize()
            .background(AppColors.bg)
    ) {
        // Dark gradient hero header with rounded bottom
        Box(
            modifier = Modifier
                .fillMaxWidth()
                .height(120.dp)
                .clip(RoundedCornerShape(bottomStart = 24.dp, bottomEnd = 24.dp))
                .background(AppColors.heroGradient),
            contentAlignment = Alignment.Center
        ) {
            Column(
                horizontalAlignment = Alignment.CenterHorizontally,
                verticalArrangement = Arrangement.spacedBy(4.dp),
                modifier = Modifier.padding(top = 8.dp, bottom = 20.dp)
            ) {
                Icon(
                    imageVector = Icons.Filled.Shield,
                    contentDescription = null,
                    tint = Color.White.copy(alpha = 0.9f),
                    modifier = Modifier.size(28.dp)
                )
                Text(
                    text = "OTP Auth PoC",
                    fontSize = 20.sp,
                    fontWeight = FontWeight.Bold,
                    color = Color.White
                )
                Text(
                    text = "Layered Anti-Bot Defenses",
                    fontSize = 12.sp,
                    fontWeight = FontWeight.Medium,
                    color = Color.White.copy(alpha = 0.6f)
                )
            }
        }

        Column(
            modifier = Modifier
                .fillMaxSize()
                .verticalScroll(rememberScrollState())
                .padding(bottom = 32.dp),
            verticalArrangement = Arrangement.spacedBy(AppSpacing.sectionGap)
        ) {
            StepIndicator(
                currentStep = currentStep,
                modifier = Modifier.padding(top = AppSpacing.sectionGap)
            )

            AnimatedContent(
                targetState = state,
                transitionSpec = {
                    (slideInHorizontally { it } + fadeIn()) togetherWith
                            (slideOutHorizontally { -it } + fadeOut())
                },
                label = "auth_state"
            ) { targetState ->
                when (targetState) {
                    is AuthState.PhoneInput, is AuthState.Error ->
                        PhoneInputView(viewModel = viewModel)
                    is AuthState.RunningPreflight, is AuthState.SolvingPoW, is AuthState.SendingOtp ->
                        LoadingView(viewModel = viewModel)
                    is AuthState.OtpSent, is AuthState.VerifyingOtp ->
                        OtpVerifyView(viewModel = viewModel)
                    is AuthState.Authenticated ->
                        SessionView(viewModel = viewModel)
                }
            }

            DefenseCard(
                modifier = Modifier.padding(horizontal = 20.dp)
            )
        }
    }
}

File: ios/OTPPoc/OTPPoc/Views/ContentView.swift
import SwiftUI

struct ContentView: View {
    @StateObject private var viewModel = AuthViewModel()

    var body: some View {
        ZStack {
            AppColors.bg.ignoresSafeArea()

            VStack(spacing: 0) {
                // Dark gradient hero header
                ZStack {
                    AppColors.heroGradient
                        .ignoresSafeArea(edges: .top)

                    VStack(spacing: 4) {
                        Image(systemName: "lock.shield.fill")
                            .font(.system(size: 28))
                            .foregroundColor(.white.opacity(0.9))
                        Text("OTP Auth PoC")
                            .font(.system(size: 20, weight: .bold))
                            .foregroundColor(.white)
                        Text("Layered Anti-Bot Defenses")
                            .font(.system(size: 12, weight: .medium))
                            .foregroundColor(.white.opacity(0.6))
                    }
                    .padding(.top, 8)
                    .padding(.bottom, 20)
                }
                .frame(height: 120)
                .clipShape(
                    RoundedCorner(radius: 24, corners: [.bottomLeft, .bottomRight])
                )

                ScrollView {
                    VStack(spacing: AppSpacing.sectionGap) {
                        // Step indicator
                        StepIndicator(currentStep: currentStep)
                            .padding(.top, AppSpacing.sectionGap)

                        // State-specific content
                        Group {
                            switch viewModel.state {
                            case .phoneInput, .error:
                                PhoneInputView(viewModel: viewModel)
                            case .runningPreflight, .solvingPoW, .sendingOtp:
                                LoadingView(viewModel: viewModel)
                            case .otpSent, .verifyingOtp:
                                OTPVerifyView(viewModel: viewModel)
                            case .authenticated:
                                SessionView(viewModel: viewModel)
                            }
                        }
                        .transition(.asymmetric(
                            insertion: .move(edge: .trailing).combined(with: .opacity),
                            removal: .move(edge: .leading).combined(with: .opacity)
                        ))
                        .animation(.spring(response: 0.4, dampingFraction: 0.8), value: viewModel.state)

                        // Defense card visible on all screens
                        DefenseCard()
                            .padding(.horizontal, 20)
                    }
                    .padding(.bottom, 32)
                }
            }
        }
    }

    private var currentStep: Int {
        switch viewModel.state {
        case .phoneInput, .error:
            return 0
        case .runningPreflight, .solvingPoW, .sendingOtp, .otpSent, .verifyingOtp:
            return 1
        case .authenticated:
            return 2
        }
    }
}

// Custom rounded corner shape for hero header
struct RoundedCorner: Shape {
    var radius: CGFloat
    var corners: UIRectCorner

    func path(in rect: CGRect) -> Path {
        let path = UIBezierPath(
            roundedRect: rect,
            byRoundingCorners: corners,
            cornerRadii: CGSize(width: radius, height: radius)
        )
        return Path(path.cgPath)
    }
}

struct LoadingView: View {
    @ObservedObject var viewModel: AuthViewModel
    @State private var dotIndex = 0

    var body: some View {
        VStack(spacing: 20) {
            HStack(spacing: 8) {
                ForEach(0..<3) { i in
                    Circle()
                        .fill(AppColors.brand500)
                        .frame(width: 10, height: 10)
                        .scaleEffect(dotIndex == i ? 1.3 : 0.7)
                        .opacity(dotIndex == i ? 1 : 0.4)
                        .animation(.easeInOut(duration: 0.4), value: dotIndex)
                }
            }
            .onAppear {
                Timer.scheduledTimer(withTimeInterval: 0.4, repeats: true) { _ in
                    dotIndex = (dotIndex + 1) % 3
                }
            }

            Text(viewModel.statusMessage)
                .font(.system(size: 14, weight: .medium))
                .foregroundColor(AppColors.textSecondary)
                .multilineTextAlignment(.center)
        }
        .padding(AppSpacing.cardPadding)
        .frame(maxWidth: .infinity)
        .background(AppColors.surface)
        .cornerRadius(AppSpacing.cardRadius)
        .shadow(color: .black.opacity(0.06), radius: 12, x: 0, y: 4)
        .padding(.horizontal, 20)
    }
}
